src/pooler.erl

```diff
@@ -220,18 +229,25 @@ take_group_member(GroupName) ->
             {_, _, X} = os:timestamp(),
             Idx = (X rem length(Pools)) + 1,
             {PoolPid, Rest} = extract_nth(Idx, Pools),
-            take_first_pool([PoolPid | Rest])
+            case {take_first_pool([PoolPid | Rest], 0), Timeout} of
```

Comment:
I did not have a lot of time to truly dig into the internals of how pooler works so if any of my assumptions are wrong please let me know:

From the looks of the code when you call the "take_first_pool" it would basically call "take_member" with a zero timeout.  If you pass in a zero timeout it will not grow the pool if it is a capacity (or if it does grow the pool, it will not wait around for a response).  The reason I do the 0 timeout first it to make sure no one pool gets too big.

Example:

Lets assume I have two pools each one connected to a different Riak server with an initial capacity of 1 and a max capacity of 100:  If I passed in the non-zero timeout first and the sudo random number being used (via the os:timestamp call) always landed on 1,  I would end up with one pool with 100 connections and 1 pool with 1 connection.  In such a case this would unbalance the load placed on the the 2 riak servers.

That said the big issue I have with this change is if you have a lot of pools (like 1000), you would be making 2000 gen_server calls just to find out there are no pools left with any space.